subsys/metrics/common/src/main/java/org/commonjava/indy/metrics/reporter/ReporterIntializer.java

```diff
@@ -0,0 +1,92 @@
+package org.commonjava.indy.metrics.reporter;
+
+import com.codahale.metrics.ConsoleReporter;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.graphite.Graphite;
+import com.codahale.metrics.graphite.GraphiteReporter;
+import org.commonjava.indy.metrics.filter.HealthCheckMetricFilter;
+import org.commonjava.indy.metrics.filter.JVMMetricFilter;
+import org.commonjava.indy.metrics.filter.SimpleMetricFilter;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.net.InetSocketAddress;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Created by xiabai on 3/3/17.
+ */
+public class ReporterIntializer {
+
+    private final static String INDY_METRICS_CONF = "org.commonjava.indy.metrics.conf";
+    private final static String INDY_METRICS_REPORTER = "org.commonjava.indy.metrics.reporter";
+    private final static String INDY_METRICS_REPORTER_GRPHITEREPORTER = "GraphiteReporter";
+    private final static String INDY_METRICS_REPORTER_GRPHITEREPORTER_HOSTNAME = "GraphiteReporter.hostname";
+    private final static String INDY_METRICS_REPORTER_GRPHITEREPORTER_PORT = "GraphiteReporter.port";
+    private final static String INDY_METRICS_REPORTER_GRPHITEREPORTER_PREFIX = "GraphiteReporter.prefix";
+    private final static String INDY_METRICS_REPORTER_GRPHITEREPORTER_SIMPLE_PERIOD = "GraphiteReporter.simple.period";
+    private final static String INDY_METRICS_REPORTER_GRPHITEREPORTER_JVM_PERIOD = "GraphiteReporter.jvm.period";
+    private final static String INDY_METRICS_REPORTER_GRPHITEREPORTER_HEALTHCHECK_PERIOD = "GraphiteReporter.healthcheck.period";
+    private final static int INDY_METRICS_REPORTER_GRPHITEREPORTER_DEFAULT_PERIOD = 30;
+
+    public static void initReporter(MetricRegistry metrics) throws Exception {
+        String conf = System.getProperty(INDY_METRICS_CONF);
```

Comment:
Instead of using system properties to initialize this, I'd rather use a configuration file so we're consistent with the rest of Indy. It makes it easier to version the configuration we're using too. Please take a look at the configurations in indy-subsys-groovy, indy-core, indy-httprox-common, etc. for examples. Configuration objects are CDI components, with a little extra stuff in there like default configuration file names and so forth. You'll find them in packages named `org.commonjava.indy.*.conf`. We provide standard configurations for the deployment tarball in src/main/conf and a default configuration file in src/main/resources.